deployment/deploymentClass.js

```diff
@@ -0,0 +1,448 @@
+        this.network,
+        this.getTemplate(this.protocol, this.network)
+      );
+    } else if (scope == "protocol") {
+      for (const network in this.data[this.protocol]["networks"]) {
+        this.generateScripts(
+          this.protocol,
+          network,
+          this.getTemplate(this.protocol, network)
+        );
+      }
+    } else if (scope == "fork") {
+      let forkProtocols = this.getAllForks();
+      for (const protocol in forkProtocols) {
+        for (const network in this.data[protocol]["networks"]) {
+          this.generateScripts(
+            protocol,
+            network,
+            this.getTemplate(protocol, network)
+          );
+        }
+      }
+    }
+  }
+
+  // Generates scripts necessary for deployment of each network
+  generateScripts(protocol, network, template) {
+    let scripts = [];
+
+    let location = this.getLocation(protocol, network);
+
+    scripts.push("rm -rf build");
+    scripts.push("rm -rf generated");
+    scripts.push("rm -rf results.txt");
+    scripts.push("rm -rf configurations/configure.ts");
+    scripts.push("rm -rf subgraph.yaml");
+
+    scripts.push(
+      "npm run prepare:yaml --PROTOCOL=" +
+        protocol +
+        " --NETWORK=" +
+        network +
+        " --TEMPLATE=" +
+        template
+    );
+    if (
+      this.data[protocol]["networks"][network]["options"][
+        "prepare:constants"
+      ] == true
+    ) {
+      scripts.push(
+        "npm run prepare:constants --PROTOCOL=" +
+          protocol +
+          " --NETWORK=" +
+          network
+      );
+    }
+    scripts.push("graph codegen");
+
+    // We don't want to deploy if we are building or just testing.
+    if (this.type == "deploy" || this.type == "") {
+      scripts.push(this.getDeploymentScript(location));
+    } else {
+      scripts.push("graph build");
+    }
+
+    this.scripts.set(location, scripts);
+  }
+
+  // Just flattens the first level of the json file so you have networks as keys, instead of protocol type.
+  flattenFirstJsonLevel() {
+    let result = {};
+    Object.keys(this.data).forEach((key1) =>
+      Object.keys(this.data[key1]).forEach(
+        (key2) => (result[key2] = this.data[key1][key2])
+      )
+    );
+    this.data = result;
+  }
+
+  // Makes sure the input arguments for the deployments you want are sensible.
+  checkValidScope() {
+    if (!this.protocol && !this.fork) {
+      throw "--PROTOCOL, --NETWORK, --FORK: Please specify a protocol, protocol and network, or fork.";
+    } else if (this.fork && (this.protocol || this.network)) {
+      throw "--PROTOCOL, --NETWORK, --FORK: If you specify a fork, you must not specify a protocol or network";
+    }
+  }
+
+  // Checks if the input arguments are valid for the deployment location.
+  checkValidService() {
+    switch (this.service) {
+      case "subgraph-studio":
+      case "studio":
+      case "s":
+      case "decentralized-network":
+      case "d":
+        return true;
```

Comment:
why have these extra `return true;` statements when you only need one at the end in this case?